mundane-io/src/main/scala/com/ambiata/mundane/io/LocalDirectory.scala

```diff
@@ -0,0 +1,193 @@
+package com.ambiata.mundane.io
+
+import com.ambiata.mundane.control._
+import com.ambiata.mundane.path._
+import java.io._
+import java.net.URI
+import scalaz._, Scalaz._, effect._
+
+/**
+ * 'LocalDirectory' represents a directory that exists on a local file system
```

Comment:
Worth adding a little bit of discussion about what "exists" really means, specifically that it is that the programmer has _explicitly_ created or checked the existence of this file, of cause something could change from underneath, but it indicates a very specific expectation and failure would occur instead of the undefined behaviour which was what was happening with the older FilePath / DirPath things.